### PR TITLE
Add recommended mod install

### LINKED ISSUE #103: Add recommended mod installs to installer
When the new installer is made, we should implement recommended additional mods somehow (Configuration Manager pretty much) as a default as everyone ends up using them anyway.

### PR DESCRIPTION
**Pull request by artehe**
Resolves #103 

Will add the option to install certain mods when installing SIT itself, to start with this will just be the Configuration Manager as most people will want this anyway.

### CODE REVIEW

**Comment by artehe** [comment]
I think this is good to go now?

File: SIT.Manager/Services/ModService.cs [hunk 1]
```diff
@@ -27,23 +28,75 @@ public class ModService(IBarNotificationService barNotificationService,
...
         }
-        Directory.CreateDirectory(Path.Combine(modsDirectory, "Extracted"));
+    }
+
+    public async Task DownloadModsCollection()
+    {
+        if (!Directory.Exists(_localModCache))
+        {
+            Directory.CreateDirectory(_localModCache);
+        }
```
Comment (Plootie) [inline SIT.Manager/Services/ModService.cs:91]: We can shorten this to just `Directory.CreateDirectory(_localModCache);` as the create directory method doesn't throw on already existing

File: SIT.Manager/ViewModels/Installation/ConfigureSitViewModel.cs [hunk 2]
```diff
@@ -174,6 +182,39 @@ private void ValidateConfiguration()
...
+
+        // Make sure that all the recommended mods are selected to start with - this isn't the best way of doing it but it works ¯\_(ツ)_/¯
+        List<ModInfo> selectedMods = [];
+        foreach (ModInfo mod in Mods)
+        {
+            if (_modService.RecommendedModInstalls.Contains(mod.Name))
+            {
+                selectedMods.Add(mod);
+            }
+        }
```
Comment (Plootie) [inline SIT.Manager/ViewModels/Installation/ConfigureSitViewModel.cs:214]: I dunno if im sleep deprived but i'm confused by this

File: SIT.Manager/Services/ModService.cs (hunk 1 above)
Comment (artehe) [reply SIT.Manager/Services/ModService.cs:91]: Good point

File: SIT.Manager/ViewModels/Installation/ConfigureSitViewModel.cs (hunk 2 above)
Comment (artehe) [reply SIT.Manager/ViewModels/Installation/ConfigureSitViewModel.cs:214]: So am I. I think I was just having a special moment 😓 

Have used something that makes more sense now.

**Review by Plootie — APPROVED** [review]
